services/explorer-indexer/src/modules/historical-sync/historical.syncer.service.ts

```diff
@@ -192,48 +217,59 @@ export default class HistoricalSyncService implements OnModuleInit {
     // @note : Right now there is no way to get the genesis transactions to set:
     // Initial validators, initial balances for delegators and faucet.
     // big TODO
-    const chuncks = [
+    const chunks = [
       {
         name: QueueJobs.NEW_PBFT_BLOCKS,
         data: {
           pbftPeriod: 0,
         },
       },
     ];
-    const foundBlockNumbers = await this.pbftService.getSavedPbftPeriods();
-    const maxVal = foundBlockNumbers.sort((a, b) => b - a)[0];
-
-    for (let i = 1; i <= maxVal; i++) {
-      if (foundBlockNumbers.indexOf(i) == -1) {
-        chuncks.push({
-          name: QueueJobs.NEW_PBFT_BLOCKS,
-          data: {
-            pbftPeriod: i,
-            type: SyncTypes.HISTORICAL,
-          } as QueueData,
-        });
+    const missingBlockNumbers = await this.pbftService.getMissingPbftPeriods();
+
+    for (const blockNumber of missingBlockNumbers) {
+      chunks.push({
+        name: QueueJobs.NEW_PBFT_BLOCKS,
+        data: {
+          pbftPeriod: blockNumber,
+          type: SyncTypes.HISTORICAL,
+        } as QueueData,
+      });
+      if (chunks.length === 1000) {
+        await this.pushChunksBulk(
+          chunks,
+          `Added periods ${chunks[0].data.pbftPeriod} to ${
+            chunks[chunks.length - 1].data.pbftPeriod
+          } to the PBFTs queue from between the first and last saved block number in the database`
+        );
       }
     }
     for (
       this.syncState.number;
       this.syncState.number < this.chainState.number;
       this.syncState.number++
     ) {
-      chuncks.push({
+      if (chunks.length === 1000) {
+        await this.pushChunksBulk(
+          chunks,
+          `Added ${chunks.length} PBFT periods in Queue up until ${
+            chunks[chunks.length - 1].data.pbftPeriod
+          }`
+        );
+      }
+      chunks.push({
         name: QueueJobs.NEW_PBFT_BLOCKS,
         data: {
           pbftPeriod: this.syncState.number,
           type: SyncTypes.HISTORICAL,
         } as QueueData,
       });
     }
-    try {
-      const added = await this.pbftsQueue.addBulk(chuncks);
-      if (added) {
-        this.logger.log(`Added ${chuncks.length} PBFT periods in Queue`);
-      }
-    } catch (error) {
-      this.logger.error(error);
-    }
+    await this.pushChunksBulk(
```

Comment:
And, at last, to treat the scenario when there isn't 1000 periods, we need to `addBulk` one last time if there's any leftovers.